state/application.go

```diff
@@ -1456,34 +1456,48 @@ func (a *Application) GetScale() int {
 // This is used on CAAS models.
 func (a *Application) ChangeScale(scaleChange int) (int, error) {
 	newScale := a.doc.DesiredScale + scaleChange
+	logger.Tracef("ChangeScale DesiredScale %v, scaleChange %v, newScale %v", a.doc.DesiredScale, scaleChange, newScale)
 	if newScale < 0 {
 		return a.doc.DesiredScale, errors.NotValidf("cannot remove more units than currently exist")
 	}
 	buildTxn := func(attempt int) ([]txn.Op, error) {
 		if attempt > 0 {
+			if err := a.Refresh(); err != nil {
+				return nil, errors.Trace(err)
+			}
 			alive, err := isAlive(a.st, applicationsC, a.doc.DocID)
 			if err != nil {
 				return nil, errors.Trace(err)
 			} else if !alive {
 				return nil, applicationNotAliveErr
 			}
-			if err := a.Refresh(); err != nil {
-				return nil, errors.Trace(err)
-			}
 			newScale = a.doc.DesiredScale + scaleChange
 			if newScale < 0 {
 				return nil, errors.NotValidf("cannot remove more units than currently exist")
 			}
 		}
-		return []txn.Op{{
+		ops := []txn.Op{{
 			C:  applicationsC,
 			Id: a.doc.DocID,
-			Assert: bson.D{{"life", Alive},
+			Assert: bson.D{
+				{"life", Alive},
 				{"charmurl", a.doc.CharmURL},
 				{"unitcount", a.doc.UnitCount},
-				{"scale", a.doc.DesiredScale}},
+				{"scale", a.doc.DesiredScale},
+			},
 			Update: bson.D{{"$set", bson.D{{"scale", newScale}}}},
-		}}, nil
+		}}
+
+		cloudSvcDoc := cloudServiceDoc{
+			DocID:                 a.globalKey(),
+			DesiredScaleProtected: true,
+		}
+		cloudSvcOp, err := buildCloudServiceOps(a.st, cloudSvcDoc)
```

Comment:
no need to pass in a struct here - just pass in a and true
cloudSvcOp, err := buildCloudServiceOps(a, true)